tests/Traits/EntityReferenceTrait.php

```diff
@@ -12,23 +14,29 @@ trait EntityReferenceTrait {
   /**
    * Get reference field in a multi-value, parsable format.
    *
-   * @param string $field_name
-   *   Reference field name.
    * @param string $entity_type
-   *   Entity type machine name.
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle.
+   * @param string $field_name
+   *   The field name.
    * @param string $labels
    *   Entity labels, comma separated.
    *
    * @return array
    *   Expanded field name with comma separated list of target IDs.
    */
-  protected function getReferenceField(string $field_name, string $entity_type, string $labels): array {
+  protected function getReferenceField(string $entity_type, string $bundle, string $field_name, string $labels): array {
+    $field_config = FieldConfig::loadByName($entity_type, $bundle, $field_name);
+    /** @var \Drupal\Core\Entity\EntityReferenceSelection\SelectionInterface $handler */
+    $handler = \Drupal::service('plugin.manager.entity_reference_selection')->getSelectionHandler($field_config);
     // Transform titles to ids and maintain the comma separated format.
     $items = explode(',', $labels);
     $items = array_map('trim', $items);
     $ids = [];
     foreach ($items as $item) {
-      $ids[] = $this->loadEntityByLabel($entity_type, $item)->id();
+      $found_entities = $handler->getReferenceableEntities($item);
```

Comment:
This is not a good way of doing that. Check for example how the taxonomy term handler implements that:

![image](https://user-images.githubusercontent.com/153362/116873207-93461100-ac17-11eb-9dcc-d623efed874a.png)

It runs all sort of checks that we don't want to have when we are simply creating an entity in our tests. For example: you might want to reference a non published term, to then test that it appears when you publish it and refresh the page.

Using that method above would make this simply impossible.

The ticket was suggesting the right way of doing this: just get the target type / bundle from the handler configuration and use it in an entity query.